### PR TITLE
Simplify datamodel

### PR DESCRIPTION
**Pull request by marcelveldt**
- Only store bare minimum of a node in the server (just the attributes raw value)
- Construct MatterNode in the client
- Construct DeviceType(instance) in the client
- Move Interview Logic 
- Bump schema version

BREAKING CHANGE

Marked as draft because this depends on updated wheels

### CODE REVIEW

File: matter_server/client/models/node.py [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+    def has_cluster(
+        self, cluster: type[_CLUSTER_T] | int, endpoint: int | None = None
+    ) -> bool:
+        """Check if node has a specific cluster on any of the endpoints."""
+        return any(
+            x
+            for x in self.endpoints.values()
+            if x.has_cluster(cluster)
+            and (x.endpoint_id == endpoint or endpoint is None)
```
Comment (MartinHjelmare) [inline matter_server/client/models/node.py:214]: Start with the None check. Then we can save one attribute access if the check short-circuits.

File: matter_server/server/device_controller.py [hunk 2]
```diff
@@ -71,18 +69,21 @@ async def initialize(self) -> None:
...
+            except (TypeError, KeyError, AttributeError) as err:
+                # only accept this to happen if the schema changed
+                if node_dict.get("interview_version") == SCHEMA_VERSION:
+                    raise err
+                # set node to None will be automatically re-interview the node
+                node = None
+            self._nodes[node_id] = node
+        # setup subscriptions and (re)interviews as task in the background
+        # as we do not want it to block our startup
+        asyncio.create_task(self._check_subscriptions_and_interviews())
```
Comment (MartinHjelmare) [inline matter_server/server/device_controller.py:86]: We need to store a reference to tasks to make sure they are not garbage collected.

Comment (marcelveldt) [reply matter_server/server/device_controller.py:86]: Also for short running tasks ? As this one reschedules itself.

Comment (balloob) [reply matter_server/server/device_controller.py:86]: yep

File: matter_server/server/device_controller.py [hunk 3]
```diff
@@ -417,47 +443,67 @@ async def _call_sdk(self, func: Callable[..., _T], *args: Any, **kwargs: Any) ->
...
+                except NodeInterviewFailed as err:
+                    LOGGER.warning(
+                        "Unable to interview Node %s, we will retry later in the background.",
+                        node_id,
+                        exc_info=err,
+                    )
+                    continue
+
+            # setup subscriptions for the node
+            if node_id not in self._subscriptions:
```
Comment (balloob) [inline matter_server/server/device_controller.py:466]: Turn this around.

```
if node_id in self._subscriptions:
    continue
```

File: matter_server/server/device_controller.py [hunk 4]
```diff
@@ -417,47 +443,67 @@ async def _call_sdk(self, func: Callable[..., _T], *args: Any, **kwargs: Any) ->
...
+
         # reschedule self to run every hour
-        self.server.loop.call_later(3600, self._schedule_interviews)
+        def _schedule():
+            asyncio.create_task(self._check_subscriptions_and_interviews())
 
-    def _schedule_interviews(self) -> None:
-        """Schedule interviews."""
-        asyncio.create_task(self._check_interviews())
+        # NOTE: do not do this in one call otherwise asyncio will start
```
Comment (balloob) [inline matter_server/server/device_controller.py:485]: Is this note really necessary? That's how asyncio works 😄

File: matter_server/server/device_controller.py [hunk 5]
```diff
@@ -337,8 +350,8 @@ def attribute_updated_callback(
             new_value = transaction.GetAttribute(path)
             LOGGER.debug("attribute updated -- %s - new value: %s", path, new_value)
```
Comment (balloob) [inline matter_server/server/device_controller.py:351]: should this log include node ID ?

File: matter_server/server/device_controller.py (hunk 4 above)
Comment (marcelveldt) [reply matter_server/server/device_controller.py:485]: hahaha, well, I've seen the mistake happen a lot.

File: matter_server/server/device_controller.py (hunk 2 above)
Comment (marcelveldt) [reply matter_server/server/device_controller.py:86]: ah good to know, I'll adjust. For my knowledge: was that changed in python3.11 ?

Comment (marcelveldt) [reply matter_server/server/device_controller.py]: adjusted

File: matter_server/server/device_controller.py (hunk 5 above)
Comment (marcelveldt) [reply matter_server/server/device_controller.py:354]: in fact that whole log line can be removed, I used it for debugging as I couldn't use a breakpoint here due to that callback called by the SDK

Comment (marcelveldt) [reply matter_server/server/device_controller.py]: Adjusted the logging a bit so the node events get logged to their own log child. Figured that logging might still come in handy

**Comment by marcelveldt** [comment]
Next step is build a new SDSK clusters wheel which includes the ALL_CLUSTERS and ALL_ATTRIBUTES lookup dicts.

**Comment by marcelveldt** [comment]
wheels are ready and I've addressed the final bugs/feedback.

**Comment by MartinHjelmare** [comment]
There are some lint failures from mypy.

**Review by balloob — APPROVED** [review]

**Pull request merged**